### PR TITLE
Remove unused six code for Python 2/3 compatibility

### PR DESCRIPTION
**Pull request by skyline75489**
This is basically a reborn version of #45254 .

Ref: #42919

### CODE REVIEW

**Comment by samestep** [comment]
Out of curiosity, why is this a separate PR rather than a revision to #45254?

**Comment by bugra** [comment]
> Out of curiosity, why is this a separate PR rather than a revision to #45254?

@samestep This is on me, I did not take the time to address some of the test failures a while ago which it looks like has been fixed/addressed since then. I am happy to close mine and we can get behind of this PR to get it merged!  

@skyline75489  Thanks for working on this! There is a test failure, can you take a look at it? I think it is related to backwards compatibility tests. Also, if you want to work on/contribute more in this line of work, we have this file that we should get rid of: https://github.com/pytorch/pytorch/blob/master/torch/_six.py

**Comment by skyline75489** [comment]
Hi @bugra! The CI failure seems to be from the upstream, not this PR . Thanks for pointing out what's next on the list for me!

**Review by mrshenli — APPROVED** [review]
Stamp for distributed package related changes.

**Comment by bugra** [comment]
@skyline75489  That sounds good! Can you retrigger the tests so that we can merge this change?

File: caffe2/python/parallel_workers.py [hunk 1]
```diff
@@ -110,7 +109,7 @@ def put_metric(self, key, value, count=True):
 
 
 class State():
-    six.add_metaclass(ABCMeta)
+    _metaclass__ = ABCMeta
```
Comment (bugra) [inline caffe2/python/parallel_workers.py:112]: There is a typo in here, it should be `__metaclass__`

**Review by bugra — CHANGES_REQUESTED** [review]
There is a typo, other than that, it looks good. Thanks for taking time on this!

File: caffe2/python/parallel_workers.py (hunk 1 above)
Comment (skyline75489) [reply caffe2/python/parallel_workers.py:112]: Fixed.

**Review by bugra — APPROVED** [review]

**Comment by bugra** [comment]
@skyline75489  Let's try to merge this.

**Comment by skyline75489** [comment]
Hi @bugra, anything else I need to do for this to land?

**Comment by bugra** [comment]
> Hi @bugra, anything else I need to do for this to land?

Can you look into test failure? Other than that, we can merge it.

**Comment by skyline75489** [comment]
There seems to be no failure after merging master. Would love to see this merged before holiday 😄

**Review by eellison — APPROVED** [review]
Stamp

**Review by gchanan — APPROVED** [review]

**Comment by bugra** [comment]
@skyline75489  Thanks for working on this, it is merged!

**Pull request closed without merging**